sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1943,7 +1955,39 @@ cl_int ExecCGCommand::enqueueImp() {
       }
     }
 
-    MQueue->getThreadPool().submit<DispatchHostTask>(DispatchHostTask(this));
+    std::vector<interop_handle::ReqToMem> ReqToMem;
+
+    if (HostTask->MHostTask->isInteropTask()) {
+      // Extract the Mem Objects for all Requirements, to ensure they are
+      // available if a user asks for them inside the interop task scope
+      const std::vector<Requirement *> &HandlerReq = HostTask->MRequirements;
+      auto ReqToMemConv = [&ReqToMem, HostTask](Requirement *Req) {
+        const std::vector<AllocaCommandBase *> &AllocaCmds =
+            Req->MSYCLMemObj->MRecord->MAllocaCommands;
+
+        for (AllocaCommandBase *AllocaCmd : AllocaCmds)
+          if (HostTask->MQueue == AllocaCmd->getQueue()) {
+            auto MemArg =
+                reinterpret_cast<pi_mem>(AllocaCmd->getMemAllocation());
+            interop_handle::ReqToMem ReqToMemEl = std::make_pair(Req, MemArg);
+            ReqToMem.emplace_back(ReqToMemEl);
+
+            return;
+          }
+
+        assert(false &&
+               "Can't get memory object due to no allocation available");
```

Comment:
The code sketched below hits this assert:

```cpp
buffer<float, 2> b_A(...);

q.submit([&](handler &h) {
   auto d_A = b_A.get_access<sycl::access::mode::write>(h);
   h.codeplay_host_task([=](sycl::interop_handle ih) {
   });
});

```


but the code seems correct to me.
Even using the accessor inside the host task hits the same assert.

Reply:
```auto d_A = b_A.get_access<sycl::access::mode::write>(h);``` This should be host_buffer, not device one.

Reply:
I believe it's due to there is still no allocation enqueued. Going to take a look at that.

Reply:
> `auto d_A = b_A.get_access<sycl::access::mode::write>(h);` This should be host_buffer, not device one.

If I wanted to use the data on the host yes, but I want the side effects of the data to be visible on the device.

Reply:
@Ruyk Done. Added the test also.

Reply:
My example works now , thanks :-)